ui/app/AppLayouts/Chat/views/ContactsColumnView.qml

```diff
@@ -250,23 +245,26 @@ Item {
                 }
             }
 
-            model: root.chatSectionModule.model
-            onChatItemSelected: root.chatSectionModule.setActiveItem(id, "")
-            onChatItemUnmuted: root.chatSectionModule.unmuteChat(id)
+            model: root.store.chatCommunitySectionModule.model
+            onChatItemSelected: {
+                root.store.chatCommunitySectionModule.setActiveItem(id, "")
+                root.store.chatCommunitySectionModule.prepareChatContentModuleForChatId(id);
```

Comment:
This line is there because it was removed from the `Component.onCompleted` in `ChatColumnView.qml`. Not sure if you understand the concept here, is to use store and set things dynamically and on demand. There is no point to create ALL chat, communities and categories instances since those are not needed to be kept alive during runtime. Ideally the Repeaters will be replaced by Loaders to optimize current implementation.

Reply:
@alexandraB99 I definitely agree that they should be moved to `Loaders`.

> There is no point to create ALL chat, communities and categories instances since those are not needed to be kept alive during runtime

I don't fully agree with this, since we're building a chat application, when user switches chat/channels/sections the state which was in the previous chat/channel (at the moment we left it) should be the same at the moment when we get back to it. And that's far easier to do if you have separate components for each chat/channel then having a single one and update it each time active chat/channel property gets changed.

What I am trying to say... if you have a chat A where you scrolled 20 messages up and have some text added in the input area and then you switch to chat B, where you start typing some reply or editing message, then when you get back to the chat A your messages list must be at the same position which was in the moment you left it (messages list scrolled 20 messages up and have entered text in the input area) and if you now get again to the chat B the state there must be the same so you may continue typing where you left off last time.

Do you agree that it's easier to do so if you have a single component for each Chat/Community section and a single component for each chat/channel?

Reply:
If this is the only reason we shall keep all these instances alive, maybe is worth looking how to store last state and open to last state when selected. However 

> But I cannot agree with this, since we're building a chat application, when user switches chat/channels/sections the state which was in the previous chat/channel (at the moment we left it) should be the same at the moment when we get back to it. And that's far easier to do if you have separate components for each chat/channel then having a single one and update it each time active chat/channel property gets changed.

This is not working now either, if I scroll up in a chat then I go to another chat and go back to the previous, the message list is again in the beginning. Also same with the chat text input, if I type something and switch to a different chat, this something is still there. Having that said, I would propose that we work on this on a different issue as it needs to be developed accordingly.

Reply:
> maybe is worth looking how to store last state and open to last state when selected

@alexandraB99 if we're going to store all those details needed for maintaining each chat/channel states and do switch chats/channels back and forth I am sure that we won't be able to put them *exactly to the same position/state* (or that will be much harder if we don't have separate components for each of them (keep in mind that different updates may happen in meanwhile)). See how that works in Discord, nothing is moved/changed even for a single pixel. Also that approach brings lot of necessary properties which will just introduce lot of unnecessary mess.

> This is not working now either, if I scroll up in a chat then I go to another chat and go back to the previous, the message list is again in the beginning

This is not done yet, but we have a good base (predisposition) for that.

> Also same with the chat text input, if I type something and switch to a different chat, this something is still there.

I tested this in the latest `base_bc`, had multiple communities plus chat section and chat/channels for all each section, added different text to each chat/channel, then switched communities, chats/channels and text I had in each of them remains there untouched.